plugins/flatpak/gs-plugin-flatpak.c

```diff
@@ -965,6 +965,9 @@ gs_plugin_flatpak_update (GsPlugin *plugin,
 			gs_flatpak_error_convert (error);
 			return FALSE;
 		}
+
+		/* Add the update applist for easier lookup */
+		gs_flatpak_transaction_add_app (transaction, app);
```

Comment:
Could you clarify why this is needed? Not sure I understand it.

Reply:
it's a flatpak transaction local cache which is useful to lookup which GsApp objects are involved in the current Transaction. The cache is populated with whatever `flatpak_transaction_list_installed_refs_for_update` returns. It's useful as in to reduce to scope of which apps we are concerned about(i.e. needing an update, runtime missing?, extensions missing?) rather than look up all installed flatpak apps and iterating over them to query that info.